mlx/primitives.cpp

```diff
@@ -2926,6 +2927,68 @@ std::vector<array> Scatter::jvp(
   throw std::runtime_error("[scatter] JVP not yet implemented");
 }
 
+std::pair<std::vector<array>, std::vector<int>> Scatter::vmap(
+    const std::vector<array>& inputs,
+    const std::vector<int>& vmap_axes) {
+  assert(inputs.size() >= 2);
+  assert(inputs.size() == vmap_axes.size());
+
+  for (size_t i = 1; i < vmap_axes.size(); i++) {
+    if (vmap_axes[i] != -1) {
+      throw std::runtime_error(
+          "[scatter]: vmap is only supported on the src input.");
+    }
+  }
+
+  const auto& src = inputs.front();
+  std::vector<array> indices(inputs.begin() + 1, inputs.end() - 1);
+  const auto& updates = inputs.back();
+
+  auto src_ax = vmap_axes.front();
+  auto upd_ax = vmap_axes.back();
+  auto scatter_axes = axes_;
+
+  auto out_ax = *std::find_if(
+      vmap_axes.begin(), vmap_axes.end(), [](int a) { return a >= 0; });
```

Comment:
Does this work if all the axes are `-1`?

Reply:
How could this happen? I added a test case with the no-op vmap (all `in_axes` equal to `-1`) and then `Primitive::vmap` is not called at all. An empty `in_axes` defaults to `{0}` so I can't see how `vmap_axes` full of `-1` could happen. I added an assertion for good measure.